### ISSUE TITLE
[ceres] build failure

### ISSUE THREAD

**Issue by aizuon** [microsoft/vcpkg#24270]
**Host Environment**
- OS: Windows 11 Build 22000
- Compiler: msvc v14.3

**To Reproduce**
`./vcpkg install ceres:x64-windows`

**Failure logs**
```
[DEBUG] Downloading https://github.com/ceres-solver/ceres-solver/archive/f68321e7de8929fbcdb95dd42877531e64f72f66.tar.gz
Error: Failed to download from mirror set:
File does not have the expected hash:
             url : [ https://github.com/ceres-solver/ceres-solver/archive/f68321e7de8929fbcdb95dd42877531e64f72f66.tar.gz ]
       File path : [ D:\vcpkg\downloads\ceres-solver-ceres-solver-f68321e7de8929fbcdb95dd42877531e64f72f66.tar.gz.19516.part ]
   Expected hash : [ b482988d837187e348250122b1acacbb4fd6354709efa6335c0322a68234a38292c072499a886b69a30614f85c818d3d2e9eeb3d3d0ca17d8f013a38d9151207 ]
     Actual hash : [ 67bbd8a9385a40fe69d118fbc84da0fcc9aa1fbe14dd52f5403ed09686504213a1d931e95a1a0148d293b27ab5ce7c1d618fbf2e8fed95f2bbafab851a1ef449 ]


[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(741):
[DEBUG] Time in subprocesses: 0 us
[DEBUG] Time in parsing JSON: 1 us
[DEBUG] Time in JSON reader: 0 us
[DEBUG] Time in filesystem: 26 us
[DEBUG] Time in loading ports: 0 us
[DEBUG] Exiting after 1.313 s (1312876 us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:84 (message):

      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.

      3. Your proxy's remote server is out of service.

      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)

      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:309 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:182 (vcpkg_download_distfile)
  ports/ceres/portfile.cmake:9 (vcpkg_from_github)
  scripts/ports.cmake:145 (include)


error: building ceres:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+ceres
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[ceres]+Build+error
including:
  package: ceres[core]:x64-windows -> 2.1.0
    vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
    vcpkg-scripts version: 7f7f1a514 2022-04-19 (19 hours ago)

Additionally, attach any relevant sections from the log files above.
```

**Additional context**
Nothing

**Comment by plumer** [comment]
I'm having the same issue. Windows 10 19043 + VS2019 16.11.12.
I have the same 'expected hash' and same 'actual hash' (and they are different).

**Comment by Thomas1664** [comment]
Will be fixed by #24156
